x-pack/plugins/security_solution/public/management/pages/trusted_apps/store/middleware.ts

```diff
@@ -38,46 +45,98 @@ const createTrustedAppsListResourceStateChangedAction = (
-    });
-
+  if (needsRefreshOfListData(store.getState())) {
     store.dispatch(
       createTrustedAppsListResourceStateChangedAction({
-        type: 'LoadedResourceState',
-        data: {
-          items: response.data,
-          totalItemsCount: response.total,
-          paginationInfo: { index: pageIndex, size: pageSize },
-        },
+        type: 'LoadingResourceState',
+        // need to think on how to avoid the casting
+        previousState: getCurrentListResourceState(store.getState()) as Immutable<
+          StaleResourceState<TrustedAppsListData>
+        >,
       })
     );
-  } catch (error) {
+
+    try {
+      const pageIndex = getListCurrentPageIndex(store.getState());
+      const pageSize = getListCurrentPageSize(store.getState());
+      const response = await trustedAppsService.getTrustedAppsList({
+        page: pageIndex + 1,
+        per_page: pageSize,
+      });
+
+      store.dispatch(
+        createTrustedAppsListResourceStateChangedAction({
+          type: 'LoadedResourceState',
+          data: {
+            items: response.data,
+            totalItemsCount: response.total,
+            paginationInfo: { index: pageIndex, size: pageSize },
+            timestamp: new Date().getTime(),
+          },
+        })
+      );
+    } catch (error) {
+      store.dispatch(
+        createTrustedAppsListResourceStateChangedAction({
+          type: 'FailedResourceState',
+          error,
+          lastLoadedState: getLastLoadedListResourceState(store.getState()),
+        })
+      );
+    }
+  }
+};
+
+const createTrustedAppDeletionSubmissionResourceStateChanged = (
+  newState: Immutable<AsyncResourceState>
+): Immutable<TrustedAppDeletionSubmissionResourceStateChanged> => ({
+  type: 'trustedAppDeletionSubmissionResourceStateChanged',
+  payload: { newState },
+});
+
+const submitDeletionIfNeeded = async (
+  store: ImmutableMiddlewareAPI<TrustedAppsListPageState, AppAction>,
+  trustedAppsService: TrustedAppsService
+) => {
+  const submissionResourceState = getDeletionSubmissionResourceState(store.getState());
+  const entryId = getDeletionDialogEntryId(store.getState());
+
+  if (isStaleResourceState(submissionResourceState) && entryId !== undefined) {
     store.dispatch(
-      createTrustedAppsListResourceStateChangedAction({
-        type: 'FailedResourceState',
-        error,
-        lastLoadedState: getLastLoadedListResourceState(store.getState()),
+      createTrustedAppDeletionSubmissionResourceStateChanged({
+        type: 'LoadingResourceState',
+        previousState: submissionResourceState,
       })
     );
+
+    try {
+      await trustedAppsService.deleteTrustedApp({ id: entryId });
+
+      store.dispatch(
+        createTrustedAppDeletionSubmissionResourceStateChanged({
+          type: 'LoadedResourceState',
+          data: null,
+        })
+      );
+      store.dispatch({
+        type: 'trustedAppDeletionDialogClosed',
+      });
+      store.dispatch({
```

Comment:
FYI - I went with a different/simpler approach - I just call `refreshList()` here directly instead of dispatching another action to the store. I will adjust my PR if you end up going with the changes above to `refreshList()`